src/appleseed/renderer/kernel/lighting/lightsamplerbase.h

```diff
@@ -83,8 +86,15 @@ class LightSamplerBase
     typedef std::vector<EmittingTriangle>       EmittingTriangleVector;
     typedef foundation::CDF<size_t, float>      EmitterCDF;
 
+    typedef std::function<void (
```

Comment:
I think we can completely get rid of the type and just pass `std::function<>` directly. We can do that because we can use `auto` at the site of declaration of the lambda.

Edit: I see that we need to pass the lambda to multiple sub-functions, so maybe it makes sense to have a dedicated type.